src/main/resources/swagger/api-docs.yaml

```diff
@@ -371,5 +379,49 @@ definitions:
           Optional bucket URI to an archive containing Jupyter notebook extension files.
           The archive must be in tar.gz format, must not include a parent directory, and
           must have an entry point named 'main'. For more information on notebook extensions,
-          see: http://jupyter-notebook.readthedocs.io/en/latest/extending/frontend_extensions.html.
+          see http://jupyter-notebook.readthedocs.io/en/latest/extending/frontend_extensions.html.
+      machineConfig:
+        description: The configurations for the machine and worker nodes
+        $ref: '#/definitions/MachineConfig'
 
+  MachineConfig:
+    description: ''
+    properties:
+      numberOfWorkers:
+        type: integer
+        description: |
+          Optional, number of workers in the cluster. For a SingleNode, this field should not be defined, as the
+          SingleNode clusters have 0 workers by default. For a Standard cluster, workers should be 2 or more.
+      masterMachineType:
+        type: string
+        description: |
+          Optional, the machine type determines the number of CPUs and memory for the master node. For example "n1-standard-16"
+          or "n1-highmem-64". If unspecified, defaults to creating a "n1-standard-4" machine. To decide which is right for you,
+          see https://cloud.google.com/compute/docs/machine-types
+      masterDiskSize:
+        type: integer
+        description: |
+          Optional, the size of the disk on the master node. Minimum size is 10GB.
+          If unspecified, default size is 500GB.
+      workerMachineType:
+        type: string
+        description: |
+          Optional, the machine type determines the number of CPUs and memory for the worker nodes. For example "n1-standard-16"
+          or "n1-highmem-64". If unspecified, defaults to creating a "n1-standard-4" machine. To decide which is right for you,
+          see https://cloud.google.com/compute/docs/machine-types
+      workerDiskSize:
+        type: integer
+        description: |
+          Optional, the size of the disk on the master node. Minimum size is 10GB.
+          If unspecified,default size is 500GB.
+      numberOfWorkerLocalSSDs:
+        type: integer
+        description: |
+          Optional, the number of local solid state disks for workers. Must be a minimum of 0. If unspecified,
+          the default number is 0
+      numberOfPreemptibleWorkers:
+        type: integer
+        description: |
+          Optional, the number of preemptible workers. Must be a minimum of 0. If unspecified,
+          the default number is 0. Should not be specified if numberOfWorkers is 0. For more information,
```

Comment:
Actually, I'll take out the line `Should not be specified if numberOfWorkers is 0`, since when `numberOfWorkers` is 0, the defaults are applied and the worker configs are ignored. 
I'll change it to `Ignored if numberOfWorkers is 0`